sdk-extensions/tracing-incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/trace/WorkerBase.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.extension.incubator.trace;
+
+import io.opentelemetry.api.metrics.BoundLongCounter;
+import io.opentelemetry.api.metrics.GlobalMeterProvider;
+import io.opentelemetry.api.metrics.LongCounter;
+import io.opentelemetry.api.metrics.Meter;
+import io.opentelemetry.api.metrics.common.Labels;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.trace.ReadableSpan;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+abstract class WorkerBase implements Runnable {
```

Comment:
It's probably ok since this is just in the incubator, but this class is confusing to me. It's unclear how one would use it in general to implement a Batching SpanProcessor. The responsibilities between the abstract class and a concrete implementation are not particularly well defined, as far as I can tell. 

I'd also prefer it if we can figure out a solution that doesn't involve inheritance. Can the functionality in this class be something that is either a) usable directly, without extending it or b) this class is concrete and takes an implementation of some strategy interface in order to do its work?

At the absolute minimum, this class would need detailed documentation on how to use it properly, as I suspect that there are many gotchas that have to be done just right for it to work correctly.

Reply:
This class can reduce the amount of copied code from `BatchSpanProcessor` worker.
Let me think about other solutions you've proposed above.

Reply:
@jkwatson I've removed `WorkerBase` class. I've introduced `WorkerExporter` which contains all methods useful for exporting a batch of spans.

Reply:
I think that `WorkerExported` might be useful compared to `WorkerBase`.